### ISSUE TITLE
Allow extensions to read NSFW state

### ISSUE THREAD

**Issue by binyaminyblatt** [mihonapp/mihon#739]
### Describe your suggested feature

Hey I apologize if somebody has requested this before I don't know how to describe it well enough to search for it so apologies if I'm repeating something 

The future request I'm looking for is to allow a third option for is NSFW which would be it might be partially NSFW but it could read the switch in the app that disables NSFW extensions and then it would if possible in the extension set all the settings for the extension to disable the NSFW parts for instance mangadex has the ability to just disable the NSFW parts and if it could read in the settings from the app itself it could then say these settings should never be allowed to be enabled and just hide them and hard code the values

As a good example mengadex has four categories and it would disable the NSFW categories and lock that setting as disabled so it could be used in an sfw mode now obviously this wouldn't work for every extension because some of them just don't have that capability 

Basically what I'm saying that when it tries to load up the class that is the extension the app could pass a parameter that is the current value of whether or not and if NSFW is allowed on this device and then based upon that a couple of if statements should be able to decide whether or not certain things are viewable or not

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.16.5](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by AntsyLich** [comment]
This is quite doable. Will keep in mind when we get around to redoing the source api

**Comment by binyaminyblatt** [comment]
Thanks 👍

**Comment by MajorTanya** [comment]
Avoid naming sources please

**Comment by binyaminyblatt** [comment]
> Avoid naming sources please

Would you like me to edit it I just left it that way so the example is easily understood

**Comment by MajorTanya** [comment]
> > Avoid naming sources please
> 
> Would you like me to edit it I just left it that way so the example is easily understood

Would be best if you did, but I understand how it is useful for the example. Just remember it for the next time you file an issue 👍

**Comment by binyaminyblatt** [comment]
> > > Avoid naming sources please
> > 
> > 
> > Would you like me to edit it I just left it that way so the example is easily understood
> 
> Would be best if you did, but I understand how it is useful for the example. Just remember it for the next time you file an issue 👍

How is that

**Comment by binyaminyblatt** [comment]
I'm just commenting because I don't want the issue to get Auto closed

**Comment by AntsyLich** [comment]
We don't do auto close bullshit

**Comment by binyaminyblatt** [comment]
I apologize

**Comment by binyaminyblatt** [comment]
Has there been any traction on this?
